bindings/aws/sns/metadata.yaml

```diff
@@ -0,0 +1,56 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: aws.sns
+version: v1
+status: alpha
+title: "AWS SNS"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/sns/
+binding:
+  output: true
+  operations:
+    - name: create
+      description: "Create a new subscription"
+capabilities: []
+builtinAuthenticationProfiles:
+  - name: "aws"
+metadata:
+  - name: topicArn
+    required: true
+    description: |
+      The SNS topic name.
+    example: '"arn:::topicarn"'
+    type: string
+  - name: region
+    required: true
+    description: |
+      The specific AWS region where the SNS topic is located.
+    example: '"us-east-1"'
+    type: string
+  - name: endpoint
+    required: false
+    description: |
+      AWS endpoint for the component to use, to connect to SNS-compatible services or emulators.
+      Do not use this when running against production AWS.
+    example: '"http://localhost:4566"'
+    type: string
+  - name: accessKey
+    required: true
+    description: |
+      The AWS Access Key to access this resource.
+    example: '"key"'
+    type: string
+  - name: secretKey
+    required: true
+    description: |
+      The AWS Secret Access Key to access this resource.
+    example: '"secretAccessKey"'
+    type: string
```

Comment:
I believe these are part of the built-in auth profile?

Reply:
oh thx for pointing out! removed `accessKey`, `secretKey` and `sessionToken`

Reply:
Correct. because this component doesn't use the "shared auth library", we need to hack it, so:

1. Remove those metadata keys because they are in the shared profile
2. Since they still are in `sns.go`, in that file add `mdignore:"true"` to the tags, such as `json:"accessKey" mapstructure:"accessKey" mdignore:"true"`

Also since you're updating `sns.go` in that file please:

1. To the `region` tag, please add: `mapstructure:"region" mapstructurealiases:"awsRegion"`
2. Remove the region from the `metadata.yaml` because `awsRegion` is already included in the shared profile

Reply:
thank you so much! updated and local test passed, should be ready now